### ISSUE TITLE
Saved Vertex not visible in current thread for Edge addition in remote storage. But works in plocal

### ISSUE THREAD

**Issue by adithyank** [orientechnologies/orientdb#3018]
Hi,

Pls see the below simple code. It adds the edge if the storage is plocal. If the storage is remote then, ov12 and ov13 is null. So, NullPointerException is coming while trying to addEdge. The version I am using is 2.0-M2

``` java
    static void addNodeAndEdge()
    {
        initFactory();
        OrientGraph og = orientGraphFactory.getTx();
        OrientVertex v12 = og.addVertex("class:invnode", "key", 12, "name", "adithyan", "age", 32);
        OrientVertex v13 = og.addVertex("class:invnode", "key", 13, "name", "adithyan", "age", 32);
        System.out.println("v12 = " + v12.getRecord().toJSON());
        System.out.println("v13 = " + v13.getRecord().toJSON());

        v12.save();
        v13.save();

        OrientVertex ov12 = (OrientVertex) og.getVertexByKey("invnode.key", 12);
        OrientVertex ov13 = (OrientVertex) og.getVertexByKey("invnode.key", 13);

        OrientEdge e = ov12.addEdge("me-ptp", ov13, "key", "e1");

        System.out.println("ov12 = " + ov12.getRecord().toJSON());
        System.out.println("ov13 = " + ov13.getRecord().toJSON());

        System.out.println("e = " + e.getRecord().toJSON());
    }
```

What is the right way of doing it. Should I use only v12 and v13 for creating Edge?
Pls advice?

**Comment by wouterv** [comment]
We just bumped into this issue. We were reworking our application using unit tests and embedded/memory database only. 

Now we starting to move toward production and half our code is broken due to this. We are using the java api to insert stuff and use queries to check some info before saving in transactions a lot (which is quite normal in java). I never anticipated that the transaction context isnt shared between client and server. 

So it is extremely painful for us to see that expecting normal behaviour, that this doesnt have any priority.

Can you perhaps tell us what needs to be done to fix this so maybe we can help.

**Comment by lvca** [comment]
We are working on this by flushing the TX to the server as soon as a remote command is executed. With current roadmap/timing/resources this will be available after 2.2, sorry.

**Comment by lvca** [comment]
Already in v3.0-M2.
